specification/app/resource-manager/Microsoft.App/stable/2024-03-01/ManagedEnvironments.json

```diff
@@ -0,0 +1,1454 @@
+              "description": "Cluster configuration which enables the log daemon to export\napp logs to a destination. Currently only \"log-analytics\" is\nsupported"
+            },
+            "zoneRedundant": {
+              "description": "Whether or not this Managed Environment is zone-redundant.",
+              "type": "boolean",
+              "x-ms-mutability": [
+                "create",
+                "read"
+              ]
+            },
+            "customDomainConfiguration": {
+              "$ref": "./CommonDefinitions.json#/definitions/CustomDomainConfiguration",
+              "description": "Custom domain configuration for the environment"
+            },
+            "eventStreamEndpoint": {
+              "description": "The endpoint of the eventstream of the Environment.",
+              "type": "string",
+              "readOnly": true
+            },
+            "workloadProfiles": {
+              "description": "Workload profiles configured for the Managed Environment.",
+              "type": "array",
+              "items": {
+                "$ref": "#/definitions/WorkloadProfile"
+              },
+              "x-ms-identifiers": [
+                "name"
+              ]
+            },
+            "kedaConfiguration": {
+              "$ref": "#/definitions/KedaConfiguration",
+              "description": "The configuration of Keda component."
+            },
+            "daprConfiguration": {
+              "$ref": "#/definitions/DaprConfiguration",
+              "description": "The configuration of Dapr component."
+            },
+            "infrastructureResourceGroup": {
+              "description": "Name of the platform-managed resource group created for the Managed Environment to host infrastructure resources. If a subnet ID is provided, this resource group will be created in the same subscription as the subnet.",
+              "type": "string",
+              "x-ms-mutability": [
+                "create",
+                "read"
+              ]
+            },
+            "peerAuthentication": {
+              "description": "Peer authentication settings for the Managed Environment",
+              "type": "object",
+              "properties": {
+                "mtls": {
+                  "description": "Mutual TLS authentication settings for the Managed Environment",
+                  "$ref": "#/definitions/Mtls"
+                }
+              }
+            },
+            "peerTrafficConfiguration": {
+              "description": "Peer traffic settings for the Managed Environment",
+              "type": "object",
+              "properties": {
+                "encryption": {
+                  "description": "Peer traffic encryption settings for the Managed Environment",
+                  "type": "object",
+                  "properties": {
+                    "enabled": {
+                      "type": "boolean",
```

Comment:
ARM recommends enums over booleans for future proof APIs. Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].

Reply:
it is the same with previous api version.